src/FondOfSpryker/Yves/CheckoutPage/Form/CheckoutBillingAddressForm.php

```diff
@@ -178,6 +200,26 @@ protected function addLastNameField(FormBuilderInterface $builder, array $option
      */
     protected function addAddress1Field(FormBuilderInterface $builder, array $options)
     {
+        $builder->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) {
+            $form = $event->getForm();
+            $data = $event->getData();
+
+            if ($form->has(static::FIELD_ADDRESS_1)) {
+                if (isset($data[static::FIELD_HOUSE_NUMBER_VALIDATION]) && $data[static::FIELD_HOUSE_NUMBER_VALIDATION] === '0') {
+                    $address1 = $data[static::FIELD_ADDRESS_1];
+                    $pattern = '~[\d]+~';
+
+                    if (preg_match($pattern, $address1) === 0) {
```

Comment:
Can you minimize the nesting here?

Reply:
Like this @DerCalli. Too much nesting is unreadable.

```php
$builder->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) {
    $form = $event->getForm();
    $data = $event->getData();

    if (!$form->has(static::FIELD_ADDRESS_1)) {
        return;
    }

    if (!isset($data[static::FIELD_HOUSE_NUMBER_VALIDATION]) || $data[static::FIELD_HOUSE_NUMBER_VALIDATION] !== '0') {
        return;
    }

    $address1 = $data[static::FIELD_ADDRESS_1];
    $pattern = '~[\d]+~';

    if (preg_match($pattern, $address1) !== 0) {
        return;
    }

    $errorMsg = $this->getFactory()
        ->getGlossaryStorageClient()
        ->translate('checkout.warning.field.housenumber', $this->getFactory()->getStore()->getCurrentLocale());

    $form->get(static::FIELD_ADDRESS_1)->addError(new FormError($errorMsg));
});
```